shapers/harfbuzz.lua

```diff
@@ -63,19 +62,44 @@ function shaper:shapeToken (text, options)
   return items
 end
 
+local _pretty_varitions = function (face)
+  local text = face.filename
+  if face.variations and face.variations ~= "" then
+    text = text.."@"..face.variations
+  end
+  local index = bitshim.band(face.index, 0xFFFF) or 0
+  local instance = bitshim.rshift(face.index, 16) or 0
+  if index or instance then
+    text = text.."["..index..","..instance.."]"
+  end
+  return text
+end
+
 -- TODO: normalize this method to accept self as first arg
 function shaper.getFace (opts)
   local face = SILE.fontManager:face(opts)
   SU.debug("fonts", "Resolved font family", opts.family, "->", face and face.filename)
   if not face or not face.filename then SU.error("Couldn't find face '"..opts.family.."'") end
   if SILE.makeDeps then SILE.makeDeps:add(face.filename) end
-  if bitshim.rshift(face.index, 16) ~= 0 then
-    SU.warn("GX feature in '"..opts.family.."' is not supported, fallback to regular font face.")
-    face.index = bitshim.band(face.index, 0xff)
+  face.variations = opts.variations
+  face.pointsize = ("%g"):format(SILE.measurement(opts.size):tonumber())
+  face.weight = ("%d"):format(opts.weight or 0)
+
+  -- Try instanciating the font, hb.instanciate() will return nil if it is not
+  -- a variable font or if instanciation failed.
+  face.tempfilename = face.filename
+  local data = hb.instanciate(face)
+  if data then
+    local tmp = os.tmpname()
+    local file = io.open(tmp, "wb")
+    file:write(data)
+    file:close()
+    face.tempfilename = tmp
+    SU.debug("fonts", "Instanciated", _pretty_varitions(face), "as", face.tempfilename)
+  elseif (face.variations ~= "") or (bitshim.rshift(face.index, 16) ~= 0) then
+    SU.debug("fonts", "Failed to instanciate", _pretty_varitions(face))
```

Comment:
We can make this an error and move the message here.

Reply:
Will do.

Reply:
I just fixed it. face.variations can be nil, I just made sure it is always set.